pkg/grpc/server_request_metadata_handler.go

```diff
@@ -0,0 +1,68 @@
+package grpc
+
+import (
+	"context"
+
+	remoteexecution "github.com/bazelbuild/remote-apis/build/bazel/remote/execution/v2"
+	"github.com/golang/protobuf/proto"
+
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/stats"
+
+	"go.opencensus.io/trace"
+)
+
+// NewRequestMetadataFetchingStatsHandler is meant to wrap ocgrpc.ServerHandler,
+// and exports additional information about the RPC from REAPI request metadata
+func NewRequestMetadataFetchingStatsHandler(base stats.Handler) stats.Handler {
+	return requestMetadataFetchingStatsHandler{base}
+}
+
+type requestMetadataFetchingStatsHandler struct {
+	stats.Handler
+}
+
+// TODO: test once there's a solution for tracking span attributes:
+// https://github.com/census-instrumentation/opencensus-go/issues/293
+
+func (rmfsh requestMetadataFetchingStatsHandler) TagRPC(ctx context.Context, rti *stats.RPCTagInfo) context.Context {
+	ctx = rmfsh.Handler.TagRPC(ctx, rti)
+
+	span := trace.FromContext(ctx)
+	if span == nil {
+		return ctx
+	}
+	if !span.IsRecordingEvents() {
+		return ctx
+	}
+
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return ctx
+	}
+
+	rmds := md.Get("build.bazel.remote.execution.v2.requestmetadata-bin")
+	if len(rmds) == 0 {
+		return ctx
+	}
+
+	var rmd remoteexecution.RequestMetadata
+
+	if err := proto.Unmarshal([]byte(rmds[0]), &rmd); err != nil {
+		return ctx
+	}
+
+	span.AddAttributes(
+		trace.StringAttribute("action_id", rmd.ActionId),
```

Comment:
Are there any conventions in OpenTracing/OpenCensus/... when it comes to namespacing attributes? Should we be using something like "RequestMetadata.action_id" instead?

I don't know a lot about this stuff, which is why I'm asking.

Reply:
I don't think there's anything obvious, the ocgrpc plugin already has stuff that's just at "root level": https://github.com/census-instrumentation/opencensus-go/blob/d7677d6af5953e0506ac4c08f349c62b917a443a/plugin/ocgrpc/trace_common.go#L90-L91